apollo-federation/src/connectors/json_selection/apply_to.rs

```diff
@@ -2828,17 +2908,221 @@ mod tests {
         );
 
         assert_eq!(
-            selection!("batch.id->map([@])->echo([@])->jsonStringify->typeof").apply_to(&json!({
-                "batch": [
-                    { "id": 1 },
-                    { "id": 2 },
-                    { "id": 3 },
-                ],
-            })),
+            selection!("batch.id->map([@])->echo([@])->jsonStringify->typeof", spec).apply_to(
+                &json!({
+                    "batch": [
+                        { "id": 1 },
+                        { "id": 2 },
+                        { "id": 3 },
+                    ],
+                })
+            ),
             (Some(json!("string")), vec![]),
         );
     }
 
+    #[rstest]
+    #[case::v0_2(ConnectSpec::V0_2)]
+    fn test_left_associative_output_shapes_v0_2(#[case] spec: ConnectSpec) {
```

Comment:
There are downsides to `rstest`: it makes it harder to run an individual test (no click the button in RustRover) and there's a tiny cost to the macros... which adds up. So when there's a single case, we should skip `rstest` (I gather this is an artifact of a previous approach, from your above comment about snapshots).


```suggestion
    fn test_left_associative_output_shapes_v0_2() {
        let spec = ConnectSpec::V0_2;
```

Reply:
https://github.com/apollographql/router/pull/7932/commits/3705ee2c9a2eea3b7b7d0e08a3c8b870f28acbae